rpcs3/Emu/RSX/rsx_cache.h

```diff
@@ -423,8 +423,7 @@ namespace rsx
 		std::string version_prefix;
 		std::string root_path;
 		std::string pipeline_class_name;
-		std::mutex fpd_mutex;
-		std::unordered_map<u64, std::vector<u8>> fragment_program_data;
+		lf_fifo<std::unique_ptr<u8[]>, 100> fragment_program_data;
```

Comment:
Last I checked this structure did not actually implement a fifo queue that did what it seemed to imply (I honestly don't remember the details, only that it was a disaster). I'll have to verify again if the implementation was somehow altered.

Reply:
I checked and it seems I modified RSXFragmentProgram enough that we do a proper deep copy now. lf_fifo is still as I remember it, so this is fine.